dev/mcal/inc/stm32f103xx.h

```diff
@@ -0,0 +1,86 @@
+/**
+ * @file stm32f103xx.h
+ * @author
+ * @brief
+ * @version 0.1
+ * @date 2024-03-09
+ *
+ * @copyright Copyright (c) 2024
+ *
+ */
+
+#ifndef DEV_MCAL_INC_STM32F103XX_H_
+#define DEV_MCAL_INC_STM32F103XX_H_
+
+#include "utils/inc/Types.h"
+
+#ifdef UNIT_TEST
+#include "baseAddress-test.h"
+#else
+#include "mcal/inc/BaseAddress.h"
+#endif  // UNIT_TEST
+
+using RegWidth_t = stm32::utils::types::RegWidth_t;
+namespace stm32 {
+namespace registers {
+namespace rcc {
+
+/********************** RCC Register Definition  **********************/
+
+struct RccRegDef{
+    union CR {
+        struct {
+            RegWidth_t HSION   :1; /* High-speed internal oscillator enable */
+            RegWidth_t HSIRDY  :1; /* High-speed internal oscillator ready */
+            RegWidth_t         :1; /* Reserved */
+            RegWidth_t HSITRIM :5; /* High-speed internal oscillator trimming */
+            RegWidth_t HSICAL  :8; /* High-speed internal oscillator calibration*/
+            RegWidth_t HSEON   :1; /* High-speed external oscillator enable */
+            RegWidth_t HSERDY  :1; /* High-speed external oscillator ready */
+            RegWidth_t HSEBYP  :1; /* High-speed external oscillator bypass */
+            RegWidth_t CSSON   :1; /* Clock security system enable */
+            RegWidth_t         :4; /* Reserved */
+            RegWidth_t PLLON   :1; /* Phase-locked loop enable */
```

Comment:
the last reserved bits should be considered here as reserved 

(always make sure summation of bit fields is equal to number of bits of whole value i.e. 'registerVal'